### PR TITLE
Split network player sync and hero save file

### PR DESCRIPTION
**Pull request by AJenbo**
This should fix items not appearing correct when inspecting a player wearing oiled items after they joined (and didn't change gear).

### CODE REVIEW

**Comment by AJenbo** [comment]
Tests aren't failing locally :(
`100% tests passed, 0 tests failed out of 170`

File: Source/pack.h [hunk 1]
```diff
@@ -67,24 +68,52 @@ struct PlayerPack {
...
-	int16_t wReserved2;   // For future use
+	uint8_t reserved2[2]; // For future use
 	uint8_t pSplLvl2[10]; // Hellfire spells
 	int16_t wReserved8;   // For future use
 	uint32_t pDiabloKillLevel;
 	uint32_t pDifficulty;
-	uint32_t pDamAcFlags; // `ItemSpecialEffectHf` is 1 byte but this is 4 bytes.
-	/**@brief Only used in multiplayer sync (SendPlayerInfo/recv_plrinfo). Never used in save games (single- or multiplayer). */
+	uint32_t pDamAcFlags;  // `ItemSpecialEffectHf` is 1 byte but this is 4 bytes.
+	uint8_t reserved3[20]; // For future use
```
Comment (qndel) [inline Source/pack.h:79]: where did 20 come from when it was 18 before?

File: Source/multi.cpp [hunk 2]
```diff
@@ -349,12 +349,10 @@ void ProcessTmsgs()
...
 {
-	PlayerPack pPack;
+	PlayerNetPack packed;
 	Player &myPlayer = *MyPlayer;
-	PackPlayer(&pPack, myPlayer, true, true);
-	pPack.friendlyMode = myPlayer.friendlyMode ? 1 : 0;
-	pPack.isOnSetLevel = myPlayer.plrIsOnSetLevel;
-	multi_send_zero_packet(pnum, cmd, reinterpret_cast<byte *>(&pPack), sizeof(PlayerPack));
+	PackNetPlayer(packed, myPlayer);
+	multi_send_zero_packet(pnum, cmd, reinterpret_cast<byte *>(&packed), sizeof(PlayerPack));
```
[the remark this answers is not in the document]
Comment (AJenbo) [reply Source/multi.cpp]: fixed, good catch :)

File: Source/pack.cpp [hunk 3]
```diff
@@ -207,109 +236,168 @@ void UnPackItem(const ItemPack &packedItem, const Player &player, Item &item, bo
...
+		player.InvGrid[i] = packed.InvGrid[i];
+
+	VerifyGoldSeeds(player);
+
+	for (int i = 0; i < MaxBeltItems; i++)
+		UnPackItem(packed.SpdList[i], player, player.SpdList[i], isHellfire);
+
+	CalcPlrInv(player, false);
+	player.wReflections = SDL_SwapLE16(packed.wReflections);
+	player.pDiabloKillLevel = SDL_SwapLE32(packed.pDiabloKillLevel);
```
Comment (StephenCWills) [inline Source/pack.cpp:308]:
```suggestion
	player.pDiabloKillLevel = SDL_SwapLE32(packed.pDiabloKillLevel);
	player.pDamAcFlags = static_cast<ItemSpecialEffectHf>(SDL_SwapLE32(packed.pDamAcFlags));
```

**Review by StephenCWills — CHANGES_REQUESTED** [review]

File: Source/pack.cpp (hunk 3 above)
Comment (AJenbo) [reply Source/pack.cpp:308]: Oh I intentionally removed this as it should be calculated by `CalcPlrInv()` and so not needed. Is that not correct?

File: Source/pack.cpp [hunk 4]
```diff
@@ -78,72 +79,100 @@ void PackItem(ItemPack &packedItem, const Item &item, bool isHellfire)
...
+	packed.bIsHellfire = gbIsHellfire ? 1 : 0;
+}
+
+void PackNetPlayer(PlayerNetPack &packed, const Player &player)
+{
+	packed.plrlevel = player.plrlevel;
+	packed.px = player.position.tile.x;
+	packed.py = player.position.tile.y;
+	CopyUtf8(packed.pName, player._pName, sizeof(packed.pName));
+	packed.pClass = static_cast<int8_t>(player._pClass);
```
[the remark this answers is not in the document]
Comment (AJenbo) [reply Source/pack.cpp]: Fixed

File: Source/pack.cpp (hunk 3 above)
Comment (StephenCWills) [reply Source/pack.cpp:308]: Oh, I get it. I just thought it was an oversight.

**Review by StephenCWills — APPROVED** [review]

File: Source/pack.cpp (hunk 3 above)
Comment (AJenbo) [reply Source/pack.cpp:308]: I should probably have made it more clear.

**Pull request merged**